hudi-client/hudi-client-common/src/main/java/org/apache/hudi/index/bucket/BucketIdentifier.java

```diff
@@ -43,7 +43,7 @@ public static int getBucketId(HoodieKey hoodieKey, String indexKeyFields, int nu
     } else {
       Map<String, String> recordKeyPairs = Arrays.stream(hoodieKey.getRecordKey().split(","))
           .map(p -> p.split(":"))
-          .collect(Collectors.toMap(p -> p[0], p -> p[1]));
+          .collect(Collectors.toMap(p -> p[0], p -> p.length == 1 ? "" : p[1]));
       hashKeyFields = Arrays.stream(indexKeyFields.split(","))
```

Comment:
How can a record key be `:::` ?

Reply:
Our users use a composite primary key （col1， col2， col3）. The value of col3 contains special characters
This problem occurs as long as the column value consists of ':'

Reply:
Can we still split by `:` if your column value has `:` here ?